File: src/main/java/com/clinichub/entity/Visit.java
package com.clinichub.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "visit")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Visit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    @Column(name = "visit_time", nullable = false)
    private java.time.LocalDateTime visitTime;

    @Column(name = "chief_complaint", length = 255)
    private String chiefComplaint;

    @Column(name = "present_illness", columnDefinition = "tinytext")
    private String presentIllness;

    @Column(name = "past_history", columnDefinition = "tinytext")
    private String pastHistory;

    @Column(name = "allergy_history", columnDefinition = "tinytext")
    private String allergyHistory;

    @Column(columnDefinition = "longtext")
    private String diagnosis;

    @Column(columnDefinition = "longtext")
    private String advice;

    @Column(name = "doctor_note", columnDefinition = "tinytext")
    private String doctorNote;

    @Column(name = "created_at", nullable = false)
    private java.time.LocalDateTime createdAt;
}

File: flutter_app/lib/models/visit.dart
import 'patient.dart';

class Visit {
  final int id;
  final Patient patient;
  final String visitTime;
  final String? chiefComplaint;
  final String? presentIllness;
  final String? pastHistory;
  final String? allergyHistory;
  final String? diagnosis;
  final String? advice;
  final String? doctorNote;
  final String createdAt;

  Visit({
    required this.id,
    required this.patient,
    required this.visitTime,
    this.chiefComplaint,
    this.presentIllness,
    this.pastHistory,
    this.allergyHistory,
    this.diagnosis,
    this.advice,
    this.doctorNote,
    required this.createdAt,
  });

  factory Visit.fromJson(Map<String, dynamic> json) => Visit(
    id: json['id'],
    patient: Patient.fromJson(json['patient']),
    visitTime: json['visitTime'],
    chiefComplaint: json['chiefComplaint'],
    presentIllness: json['presentIllness'],
    pastHistory: json['pastHistory'],
    allergyHistory: json['allergyHistory'],
    diagnosis: json['diagnosis'],
    advice: json['advice'],
    doctorNote: json['doctorNote'],
    createdAt: json['createdAt'],
  );
}
